### PR TITLE
Add assert and testcase

### LINKED ISSUE #83: Add Asserts and Testcases
due:Thursday

### PR DESCRIPTION
**Pull request by kuiktzecheng**
due:Thursday
Fixes #83

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddWorkExperienceCommandParserTest.java [hunk 1]
```diff
@@ -43,4 +43,49 @@ public void parseInvalidArgs_emptyArgs_throwsParseException() {
...
+        assertParseSuccess(parser, "  in/1    w/Intern,Google,2024  ", expectedCommand);
+    }
+
+    @Test
+    public void parseInvalidArgs_nonNumericIndex_throwsParseException() {
+        // Test input with a non-numeric index
+        assertParseFailure(parser, " in/abc w/Intern,Google,2024",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddWorkExperienceCommand.MESSAGE_USAGE));
+    }
+
```
Comment (verakohh) [inline src/test/java/seedu/address/logic/parser/AddWorkExperienceCommandParserTest.java:90]: great job adding more tests!

File: src/main/java/seedu/address/logic/parser/AddWorkExperienceCommandParser.java [hunk 2]
```diff
@@ -35,6 +35,10 @@ public AddWorkExperienceCommand parse(String args) throws ParseException {
             }
         }
 
+        // Assert that both indexString and workExperienceString are not null
```
Comment (verakohh) [inline src/main/java/seedu/address/logic/parser/AddWorkExperienceCommandParser.java:38]: nice and concise comments!

File: src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java [hunk 3]
```diff
@@ -62,9 +77,20 @@ public CommandResult execute(Model model) throws CommandException {
 
         model.setPerson(personToEdit, editedPerson);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
```
Comment (verakohh) [inline src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java:80]: good job inserting assert

File: src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java [hunk 4]
```diff
@@ -44,16 +44,31 @@ public AddWorkExperienceCommand(Index index, WorkExp workExp) {
...
+     *     the person who was updated and the added work experience.
+     * @throws CommandException If the specified index is out of bounds, indicating the person does not exist.
+     */
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
+        // Assert that the list of persons is not null
+        assert lastShownList != null : "Filtered person list should not be null";
```
Comment (verakohh) [inline src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java:61]: nice comment

File: src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java [hunk 5]
```diff
@@ -44,16 +44,31 @@ public AddWorkExperienceCommand(Index index, WorkExp workExp) {
...
+
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(MESSAGE_INVALID_INDEX);
         }
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        // Assert that the person to edit is not null
```
Comment (verakohh) [inline src/main/java/seedu/address/logic/commands/AddWorkExperienceCommand.java:69]: nice assert

**Review by verakohh — COMMENTED** [review]
lgtm!

**Review by chenle228 — APPROVED** [review]
LGTM!

**Pull request merged**